### PR TITLE
lib: make `safeRequire()` return tuple

### LINKED ISSUE #188: Remove unnecessary output from common.safeRequire()
Currently we have this when we run tests:
```
39 passing (51ms)

test/node/common-do-nothing.js ........................ 1/1 333ms
test/node/common-forward-event.js ..................... 4/4
Error: Cannot find module '__non_existing_module__'
test/node/common-safe-require.js ...................... 2/2
test/node/get-server-address-ipc.js ................... 2/2
test/node/get-server-address-tcp.js ................... 2/2
test/node/get-server-address-ws.js .................... 2/2
test/node/message-parser.js ........................... 8/8
test/node/remote-error-constructor.js ............... 20/20
test/node/remote-error-default-messages.js ............ 1/1 359ms
test/node/remote-error-from-jstp-array.js ........... 20/20
test/node/remote-error-get-jstp-array.js .............. 6/6
test/node/remote-error-to-jstp-array.js ............... 4/4
test/node/serde.js .................................. 65/65
total ............................................. 137/137
```
I think the line: `Error: Cannot find module '__non_existing_module__'` is out of place here, and I don't see any reason for this kind of message to be written to `stdout`.
I suggest we remove this.

### PR DESCRIPTION
**Pull request by nechaido**
As we discussed in #216 I add a PR that fixes #188 using different approach.
Refs:#216.
Closes: #188.

### CODE REVIEW

File: lib/record-serialization.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ module.exports = jsrs;
 // one of our priorities to optimize it.
 const USE_NATIVE_SERIALIZER = false;
 
-const jstpNative =
+const [error, jstpNative] =
   safeRequire('../build/Release/jstp') ||
```
Comment (aqrln) [inline lib/record-serialization.js]: This won't work.

File: test/node/common-safe-require.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ const common = require('../../lib/common');
 const existingModule = 'fs';
 const nonExistingModule = '__non_existing_module__';
 
-test.ok(common.safeRequire(existingModule), 'must require existing module');
+test.ok(common.safeRequire(existingModule), 'must require existing module'[1]);
 
-test.equal(common.safeRequire(nonExistingModule), null,
+test.equal(common.safeRequire(nonExistingModule)[1], null,
```
Comment (aqrln) [inline test/node/common-safe-require.js]: `[0]`

File: test/node/common-safe-require.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ const common = require('../../lib/common');
 const existingModule = 'fs';
 const nonExistingModule = '__non_existing_module__';
 
-test.ok(common.safeRequire(existingModule), 'must require existing module');
+test.ok(common.safeRequire(existingModule), 'must require existing module'[1]);
```
Comment (aqrln) [inline test/node/common-safe-require.js]: I believe you didn't mean to compare an array with 'm' ;)

Comment (lundibundi) [inline test/node/common-safe-require.js]: I think `[1]` should be in a different place =).

File: test/node/common-safe-require.js (hunk 2 above)
Comment (lundibundi) [inline test/node/common-safe-require.js]: Why don't we test this with `notOk` as the one above is tested with `ok`?

File: test/node/common-safe-require.js (hunk 3 above)
Comment (lundibundi) [inline test/node/common-safe-require.js]: That's error message =). But yeah it was funny.

Comment (aqrln) [inline test/node/common-safe-require.js]: ah, it's not a comparison, but a test message

File: lib/record-serialization.js [hunk 4]
```diff
@@ -12,9 +12,11 @@ module.exports = jsrs;
 // one of our priorities to optimize it.
 const USE_NATIVE_SERIALIZER = false;
 
-const jstpNative =
-  safeRequire('../build/Release/jstp') ||
-  safeRequire('../build/Debug/jstp');
+let [error, jstpNative] = safeRequire('../build/Release/jstp');
+
+if (error) {
+  [error, jstpNative] = safeRequire('../build/Debug/jstp');
```
Comment (belochub) [inline lib/record-serialization.js]: `error` will be overwritten here, losing the original error.

Comment (aqrln) [inline lib/record-serialization.js]: Yeah, I'd rather only show the error that occurred trying to load the release build when both failed if we should only show only one. Alternatively, we may show both, like we do now. I'm fine with it both ways, slightly leaning towards the former one.

**Comment by aqrln** [comment]
Alternative PR: https://github.com/metarhia/jstp/pull/216. Whichever wins the race will land on master.

**Comment by nechaido** [comment]
Landed in 35c6a8cbccfe7e91e5d4a9dce30df371b7448c06.

**Pull request closed without merging**